### DISCUSSION TITLE
Recursive Search For Models

### DISCUSSION THREAD

**Discussion by electric-samurai** [oobabooga/text-generation-webui#1626] [Ideas]
In my case I'd like to not have to place all my models in special locations in the model-dir.  I already have one directory with subdirectories specifying the sources of the models, sometimes a couple layers deep.  I'm working mostly with llama models - here is a hack I am using in the source to find all the models of the model-dir root and use the full path in the menu item.  You need to additional alter the source after doing this to remove explicit references to the model-dir.

```
def get_available_models():
    if shared.args.flexgen:
        return sorted([re.sub('-np$', '', item.name) for item in list(Path(f'{shared.args.model_dir}/').glob('*')) if item.name.endswith('-np')], key=str.lower)
    else:
        # return sorted([re.sub('.pth$', '', item.name) for item in list(Path(f'{shared.args.model_dir}/').rglob('*')) if not item.name.endswith(('.txt', '-np', '.pt', '.json', '.yaml'))], key=str.lower)
        return sorted([str(file_path.parent.absolute()) for file_path in Path(f'{shared.args.model_dir}/').rglob('*.bin') if not (".git" in str(file_path))], key = str.lower)
```

**Comment by JiHa-Kim** [comment]
Agree, you should submit a pull request.

**Comment by ke1ne** [comment]
Yes, please!
